Remove background grey in header

anthonydillon opened the pull request.

anthonydillon commented:
## QA
- Check that the canvas now covers the entire viewport
- Check that service blocks now go under the header and all the way to the footer
- Check there is nothing broken in the search panel

jujugui commented:
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/juju-gui/3732/
Test PASSed.

hatched commented:
QA Issues:

- There is now a line between the search input and the midpoint:
![screenshot from 2015-11-30 11 27 52](https://cloud.githubusercontent.com/assets/532033/11479042/9758b2de-9755-11e5-80ce-087448048044.png)

- Even though the content is visible through the header you can't actually click there to move anything.

I think that this branch is a little more complex than a few styling changes. Each component I believe also needs to be absolutely placed instead of there being a 'header' element that spreads across the whole top of the page.

anthonydillon commented:
I agree, will pick this up on weds and see if I can get all the elements in a row. Thanks for the QA @hatched

jujugui commented:
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/juju-gui/3757/
Test PASSed.

jujugui commented:
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/juju-gui/3759/
Test FAILed.

jujugui commented:
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/juju-gui/3760/
Test FAILed.

jujugui commented:
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/juju-gui/3764/
Test PASSed.

anthonydillon commented:
@hatched ready for another review. 

## Extra QA parts
- Moved header from the viewport to the top of the dom
- Moved one or two elements to top level nodes instead of being children
- Updated the key shortcuts panel with the new design [1]
- All z-index's now pull there index location from a z-index-list

hatched commented:
Looks good, but I still can't click a service if it's under the header which is a little odd because I can see it. What if you positioned the elements outside of any header element so that the middle area was clear and click-through-able?

anthonydillon commented:
Everything was clockmaker for me. Are you on Chrome? The midpoint element
isn't in the header. Do you mean the services when you click on search?

On Wed, 2 Dec 2015 20:17 Jeff Pihach <notifications@github.com> wrote:

> Looks good, but I still can't click a service if it's under the header
> which is a little odd because I can see it. What if you positioned the
> elements outside of any header element so that the middle area was clear
> and click-through-able?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/juju/juju-gui/pull/1028#issuecomment-161420962>.
>

hatched commented:
Yep in chrome.
- Add a service to the canvas.
- Drag that service so that the logo is in between the environment name and the search input.
- Release and click again and then drag. You'll see that you highlight the header element instead of dragging the service.

jujugui commented:
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/juju-gui/3780/
Test PASSed.

hatched commented:
This is awesome @anthonydillon :+1: QA :ok:

hatched commented on jujugui/static/gui/src/app/assets/css/base.scss:
```diff
@@ -84,6 +84,33 @@ $left-panel-width: 290px;
 $panel-space: 10px;
 
+// z-indexing
+$z-indexed-elements: search-results-list-block__tags,
```
What does this do?

hatched commented:
I need to modify some of the same files so to avoid conflicts... :shipit:

jujugui commented:
Status: merge request accepted. Url: http://ci.jujugui.org:8080/job/juju-gui-merge

jujugui commented:
Build failed: Attempt to land pull request failed
        build url: http://ci.jujugui.org:8080/job/juju-gui-merge/1284

jujugui commented:
Status: merge request accepted. Url: http://ci.jujugui.org:8080/job/juju-gui-merge
